### PR TITLE
slashing using side-channel

### PR DESCRIPTION
**Pull request by 0xAshish**

### CODE REVIEW

File: contracts/staking/slashing/SlashingManager.sol [hunk 1]
```diff
@@ -29,47 +29,35 @@ contract SlashingManager is ISlashingManager, Ownable {
...
+                /** uint256,*/
+                bytes
+            )
         );
-        require(dataList[1].toUint() == VOTE_TYPE, "Vote type is invalid");
-        require(slashingNonce < dataList[2].toUint(), "Invalid slashing nonce");
-        slashingNonce = dataList[2].toUint();
+
+        // require(slashingNonce < dataList[2].toUint(), "Invalid slashing nonce");
+        // slashingNonce = dataList[2].toUint();
```
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:47]: Shouldn't we add slashing nonce?

File: contracts/staking/slashing/SlashingManager.sol [hunk 2]
```diff
@@ -29,47 +29,35 @@ contract SlashingManager is ISlashingManager, Ownable {
...
-    ) public {
-        RLPReader.RLPItem[] memory dataList = vote.toRlpItem().toList();
-        require(
-            keccak256(dataList[0].toBytes()) == heimdallId,
-            "Chain ID is invalid"
+    function updateSlashedAmounts(bytes memory data, bytes memory sigs) public {
+        (
+            address proposer,
+            /** uint256 _slashingNonce, */
+            bytes memory _slashingInfoList
```
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:36]:
```suggestion
            uint256 _slashingNonce,
            address proposer,
            bytes memory _slashingInfoList
```

File: contracts/staking/slashing/SlashingManager.sol [hunk 3]
```diff
@@ -29,47 +29,35 @@ contract SlashingManager is ISlashingManager, Ownable {
...
+        ) = abi.decode(
+            data,
+            (
+                address,
+                /** uint256,*/
+                bytes
```
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:42]:
```suggestion
                uint256,
                address,
                bytes
```

File: contracts/staking/slashing/SlashingManager.sol [hunk 4]
```diff
@@ -29,47 +29,35 @@ contract SlashingManager is ISlashingManager, Ownable {
         heimdallId = keccak256(abi.encodePacked(_heimdallId));
     }
```
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:31]: uint256 public slashingNonce;

File: contracts/staking/slashing/SlashingManager.sol (hunk 1 above)
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:47]:
```suggestion
        slashingNonce = slashingNonce + 1;
        require(slashingNonce == _slashingNonce, "Invalid slashing nonce");
```

File: contracts/staking/slashing/SlashingManager.sol [hunk 5]
```diff
@@ -29,47 +29,35 @@ contract SlashingManager is ISlashingManager, Ownable {
...
         require(stakePower >= activeTwoByThree, "2/3+1 Power required");
-        require(dataList[0].toAddress() == proposer, "Invalid proposer");
-        require(
-            keccak256(dataList[1].toBytes()) ==
-                keccak256(abi.encodePacked(sha256(slashingInfoList))),
-            "Invalid slashInfoHash"
-        );
         //slashingInfoList[]=[[valiD,am,isJailed]]
-        uint256 slashedAmount = stakeManager.slash(slashingInfoList);
+        uint256 slashedAmount = stakeManager.slash(_slashingInfoList);
```
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:60]:
```suggestion
        uint256 slashedAmount = stakeManager.slash(slashingNonce, _slashingInfoList);
```

File: contracts/staking/slashing/SlashingManager.sol [hunk 6]
```diff
@@ -29,47 +29,37 @@ contract SlashingManager is ISlashingManager, Ownable {
...
+                address,
+                bytes
+            )
         );
-        require(dataList[1].toUint() == VOTE_TYPE, "Vote type is invalid");
-        require(slashingNonce < dataList[2].toUint(), "Invalid slashing nonce");
-        slashingNonce = dataList[2].toUint();
+
+        // require(slashingNonce < dataList[2].toUint(), "Invalid slashing nonce");
+        // slashingNonce = dataList[2].toUint();
```
Comment (jdkanani) [inline contracts/staking/slashing/SlashingManager.sol:47]: Remove this comments if we don't need.

Comment (AyushKaul) [reply contracts/staking/slashing/SlashingManager.sol:47]: maybe @0xAshish  needs it, let's wait for him.

Comment (AyushKaul) [reply contracts/staking/slashing/SlashingManager.sol:47]: maybe @0xAshish needs it, let's wait for him.

File: contracts/staking/StakingInfo.sol [hunk 7]
```diff
@@ -107,7 +107,7 @@ contract StakingInfo {
         address indexed signer
     );
     event UnJailed(uint256 indexed validatorId, address indexed signer);
-    event Slashed(uint256 indexed amount);
+    event Slashed(uint256 indexed nonce, uint256 indexed amount);
```
Comment (AyushKaul) [inline contracts/staking/StakingInfo.sol:110]: @mankenavenkatesh Slashed event changed

File: contracts/staking/slashing/SlashingManager.sol [hunk 8]
```diff
@@ -29,47 +29,31 @@ contract SlashingManager is ISlashingManager, Ownable {
...
-            keccak256(dataList[1].toBytes()) ==
-                keccak256(abi.encodePacked(sha256(slashingInfoList))),
-            "Invalid slashInfoHash"
-        );
         //slashingInfoList[]=[[valiD,am,isJailed]]
-        uint256 slashedAmount = stakeManager.slash(slashingInfoList);
+        uint256 slashedAmount = stakeManager.slash(
+            slashingNonce,
+            _slashingInfoList
+        );
```
Comment (AyushKaul) [inline contracts/staking/slashing/SlashingManager.sol]: @mankenavenkatesh changes in stakeManager.slash()

File: contracts/staking/StakingInfo.sol [hunk 9]
```diff
@@ -209,6 +204,10 @@ contract StakingInfo {
         require(registry.getStakeManagerAddress() == msg.sender);
         _;
     }
+    modifier onlySlashingManager() {
+        require(registry.getSlashingManagerAddress() == msg.sender);
```
Comment (AyushKaul) [inline contracts/staking/StakingInfo.sol:208]: please add the revert message like: "invalid sender address for slashing"

**Pull request merged**